synthesizer/snark/src/proving_key/mod.rs

```diff
@@ -58,17 +58,17 @@ impl<N: Network> ProvingKey<N> {
     #[allow(clippy::type_complexity)]
     pub fn prove_batch<R: Rng + CryptoRng>(
         locator: &str,
-        assignments: &[(ProvingKey<N>, Vec<circuit::Assignment<N::Field>>)],
+        assignments: BTreeMap<&ProvingKey<N>, &[circuit::Assignment<N::Field>]>,
```

Comment:
I would suggest changing this into an `impl Iterator` so the caller's function does not need to implement `collect` anymore.

Reply:
It's not the prettiest, but this removes a collection: 07055c3d3